nautobot/extras/api/serializers.py

```diff
@@ -494,56 +249,53 @@ class CustomLinkSerializer(ValidatedModelSerializer, NotesSerializerMixin):
 
     class Meta:
         model = CustomLink
-        fields = (
-            "url",
-            "target_url",
-            "name",
-            "content_type",
-            "text",
-            "weight",
-            "group_name",
-            "button_class",
-            "new_window",
-        )
+        fields = "__all__"
 
 
 #
 # Dynamic Groups
 #
 
 
+class DynamicGroupMembershipSerializer(ValidatedModelSerializer):
+    url = serializers.HyperlinkedIdentityField(view_name="extras-api:dynamicgroupmembership-detail")
+
+    class Meta:
+        model = DynamicGroupMembership
+        fields = "__all__"
+
+
 class DynamicGroupSerializer(NautobotModelSerializer):
     url = serializers.HyperlinkedIdentityField(view_name="extras-api:dynamicgroup-detail")
     content_type = ContentTypeField(
         queryset=ContentType.objects.filter(FeatureQuery("dynamic_groups").get_query()).order_by("app_label", "model"),
     )
     # Read-only because m2m is hard. Easier to just create # `DynamicGroupMemberships` explicitly
     # using their own endpoint at /api/extras/dynamic-group-memberships/.
-    children = NestedDynamicGroupMembershipSerializer(source="dynamic_group_memberships", read_only=True, many=True)
+    child_groups = serializers.SerializerMethodField(read_only=True)
+
+    @extend_schema_field(DynamicGroupMembershipSerializer)
+    def get_child_groups(self, obj):
+        depth = int(self.context.get("depth", 0))
+        if depth == 0:
+            return [child.id for child in obj.dynamic_group_memberships.all()]
+        else:
+            result = []
+            for child in obj.dynamic_group_memberships.all():
+                relation_info = get_relation_info_for_nested_serializers(obj, child, "dynamic_group_memberships")
+                field_class, field_kwargs = self.build_nested_field("dynamic_group_memberships", relation_info, depth)
+                result.append(field_class(child, context={"request": self.context.get("request")}, **field_kwargs).data)
+            return result
 
     class Meta:
         model = DynamicGroup
-        fields = [
-            "url",
-            "name",
-            "description",
-            "content_type",
-            "filter",
-            "children",
-        ]
+        # renamed it to child_groups here because of
+        # an if statement in select2.min.js
```

Comment:
Ew. I don't like renaming a serializer field just to work around a UI issue. 

If we really must do this, let's make sure it's documented as a change.

Reply:
Came up with a new solution in `forms.js` let me know if it is acceptable.